src/file.cpp

```diff
@@ -62,6 +62,7 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #ifdef TORRENT_DISK_STATS
 #include "libtorrent/io.hpp"
+#include "libtorrent/storage.hpp"
```

Comment:
the path will probably be in the L1 cache at that point anyway, or be pulled in shortly. I would not expect that hash to be noticeable. If you can see it in a profiler, it could probably be made a bit cheaper by using xor and shift instead of multiplication.